comp/core/workloadmeta/collectors/internal/process/process_collector.go

```diff
@@ -0,0 +1,201 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+// Package process implements the local process collector for Workloadmeta.
+package process
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"go.uber.org/fx"
+
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/errors"
+	processwlm "github.com/DataDog/datadog-agent/pkg/process/metadata/workloadmeta"
+	proccontainers "github.com/DataDog/datadog-agent/pkg/process/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	collectorID       = "local-process-collector"
+	componentName     = "workloadmeta-process"
+	cacheValidityNoRT = 2 * time.Second
+)
+
+type collector struct {
+	id      string
+	store   workloadmeta.Component
+	catalog workloadmeta.AgentType
+
+	wlmExtractor  *processwlm.WorkloadMetaExtractor
+	processDiffCh <-chan *processwlm.ProcessCacheDiff
+
+	// only used when process checks are disabled
+	processData       *Data
+	pidToCid          map[int]string
+	collectionClock   clock.Clock
+	containerProvider proccontainers.ContainerProvider
+}
+
+// NewCollector returns a new local process collector provider and an error.
+// Currently, this is only used on Linux when language detection and run in core agent are enabled.
+func NewCollector() (workloadmeta.CollectorProvider, error) {
+	wlmExtractor := processwlm.GetSharedWorkloadMetaExtractor(config.SystemProbe)
+	processData := NewProcessData()
+	processData.Register(wlmExtractor)
+
+	return workloadmeta.CollectorProvider{
+		Collector: &collector{
+			id:              collectorID,
+			catalog:         workloadmeta.NodeAgent,
+			wlmExtractor:    wlmExtractor,
+			processDiffCh:   wlmExtractor.ProcessCacheDiff(),
+			processData:     processData,
+			pidToCid:        make(map[int]string),
+			collectionClock: clock.New(),
+		},
+	}, nil
+}
+
+// GetFxOptions returns the FX framework options for the collector
+func GetFxOptions() fx.Option {
+	return fx.Provide(NewCollector)
+}
```

Comment:
I don't see this function called anywhere. 

I think it should be included in at least one catalog (probably [here](https://github.com/DataDog/datadog-agent/blob/main/comp/core/workloadmeta/collectors/all_options.go))

Else, the process collector will never be used.